lms/views/dashboard/pagination.py

```diff
@@ -0,0 +1,79 @@
+import json
+from typing import TypeVar
+from urllib.parse import parse_qs, urlencode, urlparse, urlunparse
+
+from marshmallow import ValidationError, fields, post_load
+from sqlalchemy import Select
+
+from lms.js_config_types import Pagination
+from lms.models import Assignment, Course
+from lms.validation._base import PyramidRequestSchema
+
+T = TypeVar("T", Course, Assignment)
+"""Types for which support pagination."""
+
+
+MAX_ITEMS_PER_PAGE = 100
+"""Maximum number of items to return in paginated endpoints"""
+
+
+def _get_cursor_value(items: list[T], cursor_columns: list) -> str:
+    last_element = items[-1]
+    # Get the relevant values from the last element on the page
+    values = [getattr(last_element, column.key) for column in cursor_columns]
+    return json.dumps(values)
+
+
+def _get_next_url(current_url, cursor_value) -> str:
```

Comment:
Use the current URL and just add/replace cursor instead of whitelisting parameters.